### PR TITLE
lt_xspec: Auto z and more

### PR DESCRIPTION
**Pull request by profxj**
Loads linelist and redshift if 
.z and .type='galaxy' in spectrum object.

### CODE REVIEW

File: linetools/guis/xspecgui.py [hunk 1]
```diff
@@ -75,6 +75,18 @@ def __init__(self, ispec, parent=None, zsys=None, norm=None, exten=None,
                                                 zsys=zsys, norm=norm, exten=exten,
                                                 screen_scale=self.scale,
                                                  rsp_kwargs=rsp_kwargs, **kwargs)
+        # Reset redshift from spec
+        if zsys is None:
+            if hasattr(self.spec_widg.spec, 'z'):
+                self.pltline_widg.setz(str(self.spec_widg.spec.z[self.spec_widg.select]))
```
Comment (jnburchett) [inline linetools/guis/xspecgui.py]: I tried this feature out with spectra from a specdb.spectra_from_coord() query, and the first spectrum has its redshift set, but the redshift does not change when I select a different spectrum.  Can the multiSpecWidget talk back to the main gui to change redshift when selecting different spectra?

File: linetools/guis/xspecgui.py [hunk 2]
```diff
@@ -75,6 +75,18 @@ def __init__(self, ispec, parent=None, zsys=None, norm=None, exten=None,
...
+        # Auto set line list if spec has proper object type
+        if hasattr(self.spec_widg.spec, 'stypes'):
+            if self.spec_widg.spec.stypes[self.spec_widg.select].lower() == 'galaxy':
+                self.pltline_widg.llist = ltgu.set_llist('Galaxy',in_dict=self.pltline_widg.llist)
+                self.pltline_widg.llist['Plot'] = True
+                idx = self.pltline_widg.lists.index(self.pltline_widg.llist['List'])
+                self.pltline_widg.llist_widget.setCurrentRow(idx)
+        #
```
Comment (jnburchett) [inline linetools/guis/xspecgui.py]: Loading up the GUI automatically selects the correct linelist for 'Galaxy', but the lines do not immediately show.  Instead, I have a click on a different linelist in this widget then then back to Galaxy.  At that point, I see the galaxy lines plotted.  This is quite possibly a Qt 'idiosyncracy' and perhaps works properly on other platforms.  Are the lines marked when you open up the GUI on your machine?

File: linetools/guis/xspecgui.py [hunk 3]
```diff
@@ -86,7 +98,8 @@ def __init__(self, ispec, parent=None, zsys=None, norm=None, exten=None,
 
         # Extras
         extras = QWidget()
-        extras.setMaximumWidth(130*self.scale)
+        extras.setMinimumWidth(180*self.scale)
+        extras.setMaximumWidth(280*self.scale)
```
Comment (jnburchett) [inline linetools/guis/xspecgui.py]: Thank you!

File: linetools/lists/sets/README [hunk 4]
```diff
@@ -7,3 +7,4 @@ v1.1
   Adding in X-ray lines to EUV
 v1.2
   Adding some extra forbidden lines for Galaxy (NT Jun 28, 2017)
+  Turned off distracting higher order Balmer lines in Galaxy
```
Comment (jnburchett) [inline linetools/lists/sets/README]: Once again, thanks!  Much cleaner for higher-z galaxies.

File: linetools/guis/xspecgui.py (hunk 1 above)
Comment (profxj) [reply linetools/guis/xspecgui.py]: fixed;  great catch

File: linetools/guis/xspecgui.py (hunk 2 above)
Comment (profxj) [reply linetools/guis/xspecgui.py]: nope, it is an idiosyncracy..

**Comment by profxj** [comment]
Is this ready, @jnburchett ?

**Pull request merged**